public/components/common/hooks/useUserPermissions.ts

```diff
@@ -15,25 +15,35 @@ import { WzUserPermissions } from '../../../react-services/wz-user-permissions';
 
 // It retuns user permissions
 export const useUserPermissions = () => {
-  const userPermissions = useSelector(state => state.appStateReducers.userPermissions);
-  return userPermissions;
-}
+  return useSelector((state) => state.appStateReducers.userPermissions);
+};
 
 // It returns user permissions validation and user permissions
 export const useUserPermissionsRequirements = (requiredPermissions) => {
   const userPermissions = useUserPermissions();
-  if(requiredPermissions === null){
-    return [false, userPermissions]
+  if (requiredPermissions === null) {
+    return [false, userPermissions];
   }
-  const requiredPermissionsArray = typeof requiredPermissions === 'function' ? requiredPermissions() : requiredPermissions;
-  return [WzUserPermissions.checkMissingUserPermissions(requiredPermissionsArray, userPermissions), userPermissions];
-}
+
+  if (!userPermissions) {
+    return requiredPermissions;
+  }
+
+  const requiredPermissionsArray =
+    typeof requiredPermissions === 'function' ? requiredPermissions() : requiredPermissions;
+  return [
+    WzUserPermissions.checkMissingUserPermissions(requiredPermissionsArray, userPermissions),
+    userPermissions,
+  ];
```

Comment:
This hook is returning the next interface:
```
[requiredPermissions, userPermissions]
```

but in the conditional is returned
```
requiredPermissions
```
The return in the conditional should be consistent to the interface

Reply:
If it is correct, this variable `requiredPermissions` comes by parameters an array, for ex: 
`[{action: 'agent:read', resource: 'agent:id:*'}]`

Reply:
Thanks! this is the correct return `[requiredPermissions, []]`